### PR TITLE
Show metadata for all states

### PR DESCRIPTION
**Pull request by Andrei-Stepanov**
Signed-off-by: Andrei Stepanov <astepano@redhat.com>

### CODE REVIEW

File: src/components/MetadataInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    Th,
+    Td,
+    Tbody,
+    Thead,
+    Caption,
+    TableComposable,
+} from '@patternfly/react-table';
+
+import styles from '../custom.module.css';
+import { Artifact, StateType } from '../artifact';
```
Comment (mgrabovsky) [inline src/components/MetadataInfo.tsx:42]: Unused variables.

File: src/components/ArtifactKaiState.tsx [hunk 2]
```diff
@@ -21,6 +21,8 @@
 import _ from 'lodash';
 import classNames from 'classnames';
 import * as React from 'react';
+import { useState } from 'react';
+import { useLazyQuery, useQuery } from '@apollo/client';
```
Comment (mgrabovsky) [inline src/components/ArtifactKaiState.tsx:25]: Unused variable `useQuery`.

File: src/components/ArtifactGreenwaveKaiState.tsx [hunk 3]
```diff
@@ -31,10 +32,27 @@ import {
...
+    Tabs,
+    TabsProps,
+    TabTitleIcon,
+    TabTitleText,
     Text,
     TextContent,
+    TextVariants,
 } from '@patternfly/react-core';
-import { RegisteredIcon, WeeblyIcon } from '@patternfly/react-icons';
+import { useLazyQuery, useQuery } from '@apollo/client';
```
Comment (mgrabovsky) [inline src/components/ArtifactGreenwaveKaiState.tsx:46]: Unused variable `useQuery`.

File: src/components/MetadataInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+    if (_.isEmpty(issues) || _.isNil(issues)) {
+        return null;
+    }
+    const columnNames = {
+        info: 'Info',
+        status: 'Status',
+        severity: 'Severity',
+    };
+
+    const depReadable = {
```
Comment (mgrabovsky) [inline src/components/MetadataInfo.tsx:142]: This variable is never used.

File: src/utils/artifactUtils.tsx [hunk 5]
```diff
@@ -18,14 +18,15 @@
...
-import _ from 'lodash';
+import _, { isElement, isEmpty, isNil } from 'lodash';
 import {
     CheckCircleIcon,
     ExclamationTriangleIcon,
     GhostIcon,
     HistoryIcon,
     InfoIcon,
     InProgressIcon,
+    NimblrIconConfig,
```
Comment (mgrabovsky) [inline src/utils/artifactUtils.tsx:29]: This doesn't seem to be used anywhere.

File: src/utils/artifactUtils.tsx [hunk 6]
```diff
@@ -300,6 +301,21 @@ export const getThreadID = (args: {
     return null;
 };
 
+export const getArtifactProduct = (artifact: Artifact): string | null => {
+    if (isArtifactMBS(artifact) || isArtifactRPM(artifact)) {
+        const { gate_tag_name } = artifact.payload;
+        if (isEmpty(gate_tag_name)) {
+            return null;
+        }
+        const product = gate_tag_name.match(/^.*(rhel-\d\d?)\.$/);
```
Comment (mgrabovsky) [inline src/utils/artifactUtils.tsx:310]: This only works for RHEL. Do we want to support Fedora and CentOS, as well?

File: src/utils/artifactUtils.tsx [hunk 7]
```diff
@@ -18,14 +18,15 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-import _ from 'lodash';
+import _, { isElement, isEmpty, isNil } from 'lodash';
```
Comment (mgrabovsky) [inline src/utils/artifactUtils.tsx:21]: Why these special imports all of a sudden? We always write `_.isEmpty()`, etc.

File: src/components/ArtifactGreenwaveKaiState.tsx [hunk 8]
```diff
@@ -152,6 +179,176 @@ export const FaceForGreenwaveKaiState: React.FC<
...
+                >
+                    <GreenwaveResultInfo state={state.gs} />
+                    <KaiStateMapping artifact={artifact} state={state.ks} />
+                </Tab>
+            </Tabs>
+        </>
+    );
+};
+
+export const TabWithChildren = React.forwardRef(
```
Comment (mgrabovsky) [inline src/components/ArtifactGreenwaveKaiState.tsx:342]: What is the purpose of this? It's never used anywhere.

**Review by mgrabovsky — COMMENTED** [review]
Code looks OK. I just pointed out a couple of minor things.

File: src/utils/artifactUtils.tsx (hunk 6 above)
Comment (Andrei-Stepanov) [reply src/utils/artifactUtils.tsx:310]: Gating based on brew tags is available only in RHEL.
Cenots/Fedora doesn't have this.

Comment (mgrabovsky) [reply src/utils/artifactUtils.tsx:310]: Can you please mention that in a comment?

**Comment by Andrei-Stepanov** [comment]
@mgrabovsky updated, thank you for review.

File: src/components/ArtifactGreenwaveKaiState.tsx (hunk 8 above)
Comment (Andrei-Stepanov) [reply src/components/ArtifactGreenwaveKaiState.tsx]: Removed. It was leftovers from some experiments.

**Review by mgrabovsky — APPROVED** [review]
Looks good to me.

**Pull request merged**